### PR TITLE
FORT02.04 Multicall Fix for Optimism

### PR DESCRIPTION
**Pull request by 0xtaf**
This fix removes `ethers-multicall` package and replaces it with the Multicall utils in `forta-agent-tools` and enables calls for Optimism chain.

### CODE REVIEW

File: flashloan-detector/src/helper.js [hunk 1]
```diff
@@ -140,15 +140,19 @@ module.exports = {
...
+      const contract = new MulticallContract(address, ABI);
       return contract.decimals();
     });
 
     if (decimalCalls.length > 0) {
-      const decimals = await ethcallProvider.all(decimalCalls);
+      const results = await ethcallProvider.all(decimalCalls);
+
+      if (!results[0]) {
+        return;
```
Comment (Vxatz) [inline flashloan-detector/src/helper.js:151]: I think it makes more sense to return `0` here

File: flashloan-detector/src/helper.js [hunk 2]
```diff
@@ -188,8 +192,15 @@ module.exports = {
     }
 
     if (!tokenDecimals[asset]) {
-      const contract = new Contract(asset, ABI);
-      const [decimals] = await ethcallProvider.all([contract.decimals()]);
+      const contract = new MulticallContract(asset, ABI);
+      const results = await ethcallProvider.all([contract.decimals()]);
+
+      if (!results[0]) {
+        return;
```
Comment (Vxatz) [inline flashloan-detector/src/helper.js:199]: Similarly, here maybe it's better to return max uint

**Review by Vxatz — CHANGES_REQUESTED** [review]
1) in the `calculateBorrowedAmount` function in helper.js, can you change the order of checking the `!response.data[asset]` and setting the `usdPrice`. (The correct order is the one in the `calculateNativeUsdProfit`.

2) The package version needs to be increased

**Comment by Vxatz** [comment]
One more thing, can you add retries to the 3 calls to CoinGecko? I think this help in fixing more errors
You could add 2 max retries and set a timeout of 500ms between each one

**Comment by 0xtaf** [comment]
made some substantial changes with the coingecko calls

**Review by Vxatz — APPROVED** [review]
Approved ✅

**Pull request merged**